### ISSUE TITLE
Fixture django_db_setup stopped working when upgrading from 4.6 to 4.7

### ISSUE THREAD

**Issue by jeroenbrouwer** [pytest-dev/pytest-django#1115]
I've been using a custom `django_db_setup` fixture which has worked for years. After upgrading from `4.6` to `4.7` this suddenly doesn't work anymore, resulting in the following error:

`RuntimeError: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enable it.`

This might be a bug. Anybody any idea how to fix this from `4.7`?
The fixture is something like this:

```
@pytest.yield_fixture(scope="session", autouse=True)
def django_db_setup(django_db_blocker):
    with django_db_blocker.unblock():
        # Code that requires DB access and populates data required for all tests.
        Model.objects.create()

        # At this point the DB setup for the tests is complete
        yield
```

**Comment by dperetti** [comment]
Shouldn't your `yield` be aligned with the `with`?
```
@pytest.yield_fixture(scope="session", autouse=True)
def django_db_setup(django_db_blocker):
    with django_db_blocker.unblock():
        # Code that requires DB access and populates data required for all tests.
        Model.objects.create()

    # At this point the DB setup for the tests is complete
    yield
```

**Comment by jeroenbrouwer** [comment]
Thanks for the suggestion, it doesn't solve it unfortunately.

**Comment by jeroenbrouwer** [comment]
Somehow it always worked without marking the individual tests with `pytest.mark.django_db` until now.... Fixed for me, so I'm closing this issue.

**Comment by jeroenbrouwer** [comment]
Reopening because the problem still occurs when running this in my CI pipeline. Downgrading to `4.6` fixes it.

**Comment by yhay81** [comment]
After updating pytest-django from v4.4 to v4.8, I'm experiencing the same database access error despite using @pytest.mark.django_db.
This issues also seems to be the same one. https://github.com/pytest-dev/pytest-django/issues/1118

**Comment by bluetech** [comment]
It does work for me, so I would need some way to reproduce it in order to help.

**Comment by bluetech** [comment]
One note, `pytest.yield_fixture` is deprecated, you should change it to `pytest.fixture`. Also, you should just remove the `yield`, it's not really needed (you don't need a hook wrapper, just a regular hook). This might actually help.

**Comment by bluetech** [comment]
Argh sorry I didn't mean hook/hookwrapper, I meant fixture/yield fixture.  You should write it like this:

```py
@pytest.fixture(scope="session", autouse=True)
def django_db_setup(django_db_setup, django_db_blocker):
    with django_db_blocker.unblock():
        # Code that requires DB access and populates data required for all tests.
        Model.objects.create()
```

I added a `django_db_setup` parameter, and removed the `yield`.

**Comment by jeroenbrouwer** [comment]
> Argh sorry I didn't mean hook/hookwrapper, I meant fixture/yield fixture. You should write it like this:
> 
> ```python
> @pytest.fixture(scope="session", autouse=True)
> def django_db_setup(django_db_setup, django_db_blocker):
>     with django_db_blocker.unblock():
>         # Code that requires DB access and populates data required for all tests.
>         Model.objects.create()
> ```
> 
> I added a `django_db_setup` parameter, and removed the `yield`.

Thanks for the notice about `yield_fixture` being deprecated. According to the docs it's just an alias of `fixture`, so changing that shouldn't fix anything. The `fixture` decorator still supports `yield` and I'm using that to perform some cleanup after the `yield`. Not sure how I would go about the cleanup otherwise?

**Comment by bluetech** [comment]
Yes, it's probably fine, just something to try. The important thing is to make sure to request `django_db_setup` from your own `django_db_setup`.

**Comment by jeroenbrouwer** [comment]
> Yes, it's probably fine, just something to try. The important thing is to make sure to request `django_db_setup` from your own `django_db_setup`.

The thing is that it always worked, but stopped working with version `4.7+` and only in my CI pipelines... For now I'm just staying on `4.6` as this fixture still works on that version.
